packages/firestore/src/core/event_manager.ts

```diff
@@ -266,18 +244,19 @@ export class QueryListener {
   }
 
   private shouldRaiseEvent(snap: ViewSnapshot): boolean {
-    // We don't need to handle includeDocumentMetadataChanges here because
```

Comment:
Note: This is no longer true since we don't strip out the metadata-only changes beforehand (users could still ask for them even if the query-level metadata changes are turned off)

Furthermore, this code snippet uses a loop instead of find() for better IE support.

Reply:
FWIW we currently shim `Array.prototype.find` for environments that don't support it (see https://github.com/firebase/firebase-js-sdk/blob/master/packages/polyfill/index.ts#L24-L25).

If it improves readability please don't hesitate to use it, however in this case, it's a freebie 😄

Reply:
Hrm.  Thanks for flagging this...

This is another divergence from the Android / iOS implementations which still filter out the metadata-only changes, and I think filtering them out may make sense.  I don't think it's useful to call .docChanges({includeMetadataChanges: true}) unless you also specified {includeMetadatChanges: true} in onSnapshot()...  since you're not guaranteed to see every metadata-only change that way.  It would be kind of random.  We could even detect that and throw an exception ("If you pass includeMetadatChanges: true to docChanges() you must also pass it when calling onSnapshot()") if we wanted to be extra helpful.

WDYT?

Reply:
I was kinda torn between this and throwing an error. I don't think we ever discussed this before and I noticed that this wasn't really addressed in the original PR. I think silently ignoring this option seems like an oversight, and I changed it to throw an error for now. The only benefit of supporting it the way I did originally was to try to do what the user asked for, but it does indeed create some rather strange semantics.